tests/dispatch/test_proxy_http.py

```diff
@@ -105,23 +116,36 @@
         backend=raw_io,
         proxy_mode=httpx.HTTPProxyMode.TUNNEL_ONLY,
     ) as proxy:
-        response = await proxy.request("GET", f"https://example.com")
+        response1 = await proxy.request("GET", f"https://example.com")
 
-        assert response.status_code == 404
-        assert response.headers["Server"] == "origin-server"
+        assert response1.status_code == 404
+        assert response1.headers["Server"] == "origin-server"
 
-        assert response.request.method == "GET"
-        assert response.request.url == "https://example.com"
-        assert response.request.headers["Host"] == "example.com"
+        assert response1.request.method == "GET"
+        assert response1.request.url == "https://example.com"
+        assert response1.request.headers["Host"] == "example.com"
+        await response1.read()
```

Comment:
Had a question here, is there a way that we can detect that a response doesn't need data (ie `Content-Length: 0` or a status code that must have no content) and to immediately "release" the response?

Reply:
As in, is there a way to make calling `await .read()` optional if we *know* it has no content, for the sake of convenience? I think checking for `Content-Length: 0` would be feasible/sensible, but not sure it's worth breaking the consistency.

To be rigorous I think we should really be using an `async with`-based API on async dispatchers, something like…

```python
async with proxy.request("GET", "https://example.com") as response:
    # response already .read().
    # It wouldn't be if we passed `stream=True`.
    # It is automatically `.close()`d on aexit.
```

(Right now this design is hard to implement because nothing follows this pattern internally or in our public API… Anyway, just jotting down thoughts here, kind of related to thoughts I had after #352.)

Reply:
Yeah I don't know, I left it as `.read()` for now. Understandable why we don't want the special cases.